### PR TITLE
drivers/cc110x: use netdev_register

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description
I added  the call to `netdev_register()` in `cc110x_setup()`.

### Testing procedure

The test in `tests/driver_cc110x` runs.
```
> ifconfig
2020-12-01 10:19:39,170 #  ifconfig
2020-12-01 10:19:39,173 # Iface  5  HWaddr: 22  Channel: 0 
2020-12-01 10:19:39,178 #            TX-Power: 0dBm L2-PDU:253  MTU:1280  HL:64  RTR  
2020-12-01 10:19:39,180 #           6LO  IPHC  
2020-12-01 10:19:39,183 #           Source address length: 1
2020-12-01 10:19:39,186 #           Link type: wireless
2020-12-01 10:19:39,191 #           inet6 addr: fe80::ff:fe00:22  scope: link  VAL
2020-12-01 10:19:39,194 #           inet6 group: ff02::2
2020-12-01 10:19:39,197 #           inet6 group: ff02::1
2020-12-01 10:19:39,200 #           inet6 group: ff02::1:ff00:22
2020-12-01 10:19:39,201 #           
2020-12-01 10:19:39,204 #           Statistics for Layer 2
2020-12-01 10:19:39,207 #             RX packets 0  bytes 0
2020-12-01 10:19:39,211 #             TX packets 2 (Multicast: 2)  bytes 44
2020-12-01 10:19:39,214 #             TX succeeded 2 errors 0
2020-12-01 10:19:39,217 #           Statistics for IPv6
2020-12-01 10:19:39,220 #             RX packets 0  bytes 0
2020-12-01 10:19:39,225 #             TX packets 2 (Multicast: 2)  bytes 112
2020-12-01 10:19:39,228 #             TX succeeded 2 errors 0
2020-12-01 10:19:39,228 # 

```

### Issues/PRs references

To benefit from `gnrc_netif_get_by_type()` in #15120

### CODE REVIEW

**Comment by fabian18** [comment]
@maribu

**Review by maribu — COMMENTED** [review]
Looks good. While you're at it, could you also add `luid_netdev_get(const netdev_t *netdev, void *addr, size_t addr_len)`? That should work similar to `luid_netdev_get_eui48()` and `luid_netdev_get_eui64()`, but without constraints on the generated address and its size.

This in turn could be used to generate stable L2 addresses for the CC1101. There is one pitfall though: The address `CC1XXX_BCAST_ADDR` is used for broadcast and must not be used as L2 address. I'm not sure if instead adding an `luid_netdev_get_c1xxx_addr()` that is guarantneed to generate suitable 1 byte addresses is better. Since there currently is only the cc110x driver in the cc1xxx family, it might be better handled in the driver directly.

The define `CC110X_L2ADDR_AUTO` and the `struct` member `cc110x_params_t::l2addr` will no longer be needed with this.

File: drivers/include/net/netdev.h [hunk 1]
```diff
@@ -280,6 +280,7 @@ typedef enum {
     NETDEV_ANY = 0,         /**< Will match any device type */
     NETDEV_AT86RF215,
     NETDEV_AT86RF2XX,
+    NETDEV_CC110X,
```
[the remark this answers is not in the document]
Comment (benpicco) [reply drivers/include/net/netdev.h:283]: Yes, I've made this mistake before as well.
And please add a warning on top, that while this list may *seem* alphabetically sorted, new entries must be added at the bottom so existing IDs don't change.

**Comment by fabian18** [comment]
> While you're at it, could you also add luid_netdev_get(const netdev_t *netdev, void *addr, size_t addr_len)? 

But in another PR, right?

**Review by maribu — APPROVED** [review]
ACK. Thanks for addressing this

**Comment by maribu** [comment]
Please squash. I think two commits should be enough: One for adding the CC110X netdev type, one for making use of it in `netdev_register()`.

**Comment by fabian18** [comment]
Ok, let me squash

**Comment by fabian18** [comment]
> While you're at it, could you also add luid_netdev_get(const netdev_t *netdev, void *addr, size_t addr_len)? That should work similar to luid_netdev_get_eui48() and luid_netdev_get_eui64(), but without constraints on the generated address and its size.


Just like that?
```C

void luid_netdev_get(const netdev_t *netdev, void *addr, size_t addr_len) {
    luid_base(addr, sizeof(*addr));
#ifdef MODULE_NETDEV_REGISTER
    if (addr_len) {
        ((uint8_t *)addr)[addr_len - 1] ^= netdev->index;
        if (addr_len > 1) {
            ((uint8_t *)addr)[addr_len - 2] ^= netdev->type;
        }
    }
#else
    (void)netdev;
#endif
}
```
How to handle the case if `addr_len == 1`? Is it a problem that `netdev->type` has no influence then?

> I'm not sure if instead adding an luid_netdev_get_c1xxx_addr() that is guarantneed to generate suitable 1 byte addresses is better.

Very driver specific to me. Could `luid_netdev_get()` instead repeat some procedure (which I don´t know yet) until the address is not all `0x00` and not all `0xFF` because I doubt that any other address will be used for broadcast.

**Comment by maribu** [comment]
Looks good, except `sizeof(*addr)` will not work. You can use `addr_len` instead.

IMO you don't need to check for `addr_len == 0`, but rather just do `asssert((addr != NULL) && (addr_len > 0));`. I don't think that production code should ever call this with `addr_len == 0`.

> How to handle the case if `addr_len == 1`? Is it a problem that `netdev->type` has no influence then?

I'm not sure. There currently is only one type of network devices that uses single byte L2 addresses now, and new devices will likely have at least 2 byte addresses for scalability reasons. But there are the CC1200 transceivers for which a driver is being developed, which also as single byte addresses. (It is the driver for which the `cc1xxx_common` module was split out of the `cc110x` driver.) I think that PR hasn't seen any activity in a while, but maybe a driver for that does make it to RIOT eventually.

If we want to play it safe, we could go e.g. with

```C
((uint8_t *)addr)[0] ^= netdev->index ^ (netdev->type << 4);
```

Taking only the four least significant bits of the type should be good enough to get locally unique L2 address. (And unique L2 addresses are only mandatory if the transceivers have the same PHY/DLL. But it is still very convenient, as this translates to locally unique IPv6 addresses for each interface very easily.)

I think we should avoid emitting code for handling the `addr_len == 1` case, unless the oddity of the CC1101 is actually used. E.g. like this:

```C
assert(IS_USED(MODULE_CC110X) || (addr_len > 1));
/* Only the CC1100/CC1101 transceivers use single byte addresses,
 * so only emit code for that special case if the driver is
 * actually used */
if (IS_USED(MODULE_CC110X) && (addr_len == 1)) {
    ((uint8_t *)addr)[0] ^= netdev->index ^ (netdev->type << 4);
}
```

**Pull request merged**